Update CHANGELOG (v0.16.0) (#204)
RobotecAI/RobotecGPULidar: CHANGELOG.md

# Change Log

> ## [0.16.0] 7 November 2023

Author: ```suggestion
## [0.16.0] 10 November 2023
```

# Change Log

## [0.16.0] 7 November 2023

### Added

> - Added relative node's priority, which determines their execution order
>   - Added API calls to set/get node's priority:
>     - `rgl_graph_node_set_priority`
>     - `rgl_graph_node_get_priority`

Reviewer: ```suggestion
- Added API calls to set relative node's priority, which determines their execution order within a graph (but not between graphs).
  - `rgl_graph_node_set_priority`
  - `rgl_graph_node_get_priority`
```

### Changed

> - Changed the way RGL runs graphs
>   - It is performed fully asynchronous now

Reviewer: ```suggestion
- `rgl_graph_run(...)` is now fully asynchronous, i.e. it will initiate computations without waiting for them to finish
```

> - Moved Entity ID from SBT to OptixInstance
>   - Efficiency has been improved
>   - Maximum value has been reduced
>     - The Entity ID is now held in a 28-bit integer (OptiX limit) instead of a 32-bit integer

Reviewer: ```suggestion
- Changed internal implementation of Entity ID feature
  - The Entity ID is now held in a 28-bit integer (OptiX limit) instead of a 32-bit integer
```

> - Renamed XYZ field name to be more precise
>   - `RGL_FIELD_XYZ_F32` -> `RGL_FIELD_XYZ_VEC3_F32`

Reviewer: ```suggestion
- Renamed XYZ field name to be more precise and compatible with future features
  - `RGL_FIELD_XYZ_F32` -> `RGL_FIELD_XYZ_VEC3_F32`
```
